### PR TITLE
DEVPROD-940: Add SetLastRevision mutation

### PR DESCRIPTION
**Pull request by minnakt**
DEVPROD-940

### Description
Note: the file diff looks big but 611 lines were added through code generation and these lines can be ignored.

This PR adds the `SetLastRevision` mutation which can be used to fix the repotracker error of a given project. It also adds the `user.projectPermission` query which can be used to check a user's permissions for a given project.

### Testing
- Added GraphQL tests
- On staging:
   -  I changed the Sandbox project to a bad revision through the DB, and then I force ran the repotracker job.
   -  Running the repotracker job generated a repotracker error for the Sandbox project (screenshot below).
   - I fixed the repotracker error through the `SetLastRevision` mutation. I force ran the repotracker job again to make sure there were no errors.
staging

### CODE REVIEW

File: graphql/mutation_resolver.go
Comment (sophstad) [inline graphql/mutation_resolver.go:1]: Grrr why are these diffs being collapsed again 😭

File: graphql/mutation_resolver.go [hunk 1]
```diff
@@ -716,6 +716,39 @@ func (r *mutationResolver) SaveRepoSettingsForSection(ctx context.Context, repoS
 	return changes, nil
 }
 
+// SetLastRevision is the resolver for the setLastRevision field.
+func (r *mutationResolver) SetLastRevision(ctx context.Context, opts SetLastRevisionInput) (*SetLastRevisionPayload, error) {
+	if len(opts.Revision) < gitHashLength {
+		return nil, InternalServerError.Send(ctx, "insufficent characters provided for revision")
```
Comment (malikchaya2) [inline graphql/mutation_resolver.go:722]: nit: should we maybe include the length it should be?

**Review by malikchaya2 — APPROVED** [review]
Approved with nit

**Review by sophstad — APPROVED** [review]

**Comment by minnakt** [comment]
parsley test keeps flaking but is tracked by [DEVPROD-1904](https://jira.mongodb.org/browse/DEVPROD-1904) so i will merge

**Pull request merged**